x/payment/client/testutil/integration_test.go

```diff
@@ -131,11 +131,16 @@ func (s *IntegrationTestSuite) TestSubmitWirePayForMessage() {
 				require.Equal(tc.expectedCode, txResp.Code,
 					"test: %s, output\n:", tc.name, out.String())
 
+				// event1 = sdk.Event{
+				// 	Type: types.EventTypePayForMessage,
+				// 	Attributes: []abci.EventAttribute{},
+				// }
 				events := txResp.Logs[0].GetEvents()
 				for _, e := range events {
 					switch e.Type {
-					case types.AttributeKeySpender:
-						s.Equal(types.EventTypePayForMessage, e)
+					case types.EventTypePayForMessage:
+						//s.Equal(clientCtx.GetFeeGranterAddress(), sdk.AccAddress(e.GetAttributes()[0].Value))
+						s.Equal(out.dataString(), e.GetAttributes()[1].Value)
```

Comment:
How can we get the spender address in this test to compare with the event output?

Reply:
That's a good question, I'm not exactly sure which account the integration test is using, but that's not super important, what is important is that there is a viable address there, so we could simply iterate through the attributes of the event and find "spender" (or rename to "signer" like described in another comment.

```go
_, err = sdk.AccAddressFromBech32(signer)
assert.NoError(t, err)
```

However, we could test to see that the message size is what it should be as we know that.